core/src/main/java/hudson/model/AutoCompletionCandidates.java

```diff
@@ -173,4 +173,23 @@ private String contextualNameOf(Item i) {
     private static boolean startsWithImpl(String str, String prefix, boolean ignoreCase) {
         return ignoreCase ? StringUtils.startsWithIgnoreCase(str, prefix) : str.startsWith(prefix);
     }
+
+    /**
+     * Auto-completes possible job names with support for folder suggestions.
+     * Inspired from hudson.plugins.parameterizedtrigger.BuildTriggerConfig
+     * @param value The value the user has typed in.
+     * @param container The nearby contextual {@link ItemGroup} to resolve relative job names from.
+     */
+
+    public static AutoCompletionCandidates ofJobNames(final String value, ItemGroup container) {
+        AutoCompletionCandidates candidates = new AutoCompletionCandidates();
+        List<Job> jobs = Jenkins.get().getAllItems(Job.class);
```

Comment:
`getAllItems` can be quite expensive to run.

Reply:
This might be a good candidate for getAllItems(Class, Predicate). It does iterate every job but it checks the predicate condition before permissions. Note getAllItems and friends already does a hasPermission check for Item.READ so the additional check below is not required.